Sources/TokamakCore/Views/Image.swift

```diff
@@ -17,27 +17,121 @@
 
 import Foundation
 
-public struct Image: _PrimitiveView {
-  let label: Text?
+public class AnyImageProviderBox: AnyTokenBox, Equatable {
+  public struct _Image {
+    public indirect enum Storage {
+      case named(String, bundle: Bundle?)
+      case resizable(Storage, capInsets: EdgeInsets, resizingMode: Image.ResizingMode)
+    }
+
+    public let storage: Storage
+    public let label: Text?
+  }
+
+  public static func == (lhs: AnyImageProviderBox, rhs: AnyImageProviderBox) -> Bool {
+    lhs.equals(rhs)
+  }
+
+  public func equals(_ other: AnyImageProviderBox) -> Bool {
+    fatalError("implement \(#function) in subclass")
+  }
+
+  public func resolve(in environment: EnvironmentValues) -> _Image {
+    fatalError("implement \(#function) in subclass")
+  }
+}
+
+private class NamedImageProvider: AnyImageProviderBox {
   let name: String
   let bundle: Bundle?
+  let label: Text?
 
-  public init(_ name: String, bundle: Bundle? = nil) {
-    label = Text(name)
+  init(name: String, bundle: Bundle?, label: Text?) {
     self.name = name
     self.bundle = bundle
+    self.label = label
   }
 
-  public init(_ name: String, bundle: Bundle? = nil, label: Text) {
-    self.label = label
-    self.name = name
-    self.bundle = bundle
+  override func equals(_ other: AnyImageProviderBox) -> Bool {
+    guard let other = other as? NamedImageProvider else { return false }
+    return other.name == name
+      && other.bundle?.bundlePath == bundle?.bundlePath
+      && other.label == label
   }
 
-  public init(decorative name: String, bundle: Bundle? = nil) {
-    label = nil
-    self.name = name
-    self.bundle = bundle
+  override func resolve(in environment: EnvironmentValues) -> ResolvedValue {
+    .init(storage: .named(name, bundle: bundle), label: label)
+  }
+}
+
+private class ResizableProvider: AnyImageProviderBox {
```

Comment:
A bit unsure about this class hierarchy, are we going to have more than one subclass of `AnyImageProviderBox`?

Reply:
We will likely have something like `SystemImageProvider` in the future.

Reply:
Ah, you mean for something like SF Symbols?

Reply:
Yeah, exactly.

Reply:
Those will need to be resolved in the environment so they can access the font and accessibility settings, so I think it makes sense to have it be a token.